src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/Conversions.cs

```diff
@@ -41,6 +41,22 @@ protected override ConversionsBase WithNullabilityCore(bool includeNullability)
             return new Conversions(_binder, currentRecursionDepth, includeNullability, this);
         }
 
+        public static void CheckDefaultParameterMatch(ErrorCode code, ParameterSymbol source, ParameterSymbol target, int paramIdx, Location location, BindingDiagnosticBag diagnostics)
+        {
+            Debug.Assert(code == ErrorCode.ERR_OptionalParamValueMismatch || code == ErrorCode.WRN_OptionalParamValueMismatch);
```

Comment:
Nit: could do `code is ErrorCode.ERR_OptionalParamValueMismatch or ErrorCode.WRN_OptionalParamValueMismatch`